_docs-sources/pipelines/overview/configuration.md

```diff
@@ -1,87 +1,360 @@
+
+In most circumstances, the same role would be assumed by multiple units of configuration within a repository (e.g. all units within a given directory configure resources for the same AWS account). In that circumstance, it would be more appropriate not to define a `gruntwork.hcl` file at all, and instead define a `.gruntwork` directory at the root of the repository with a file ending in `.hcl` that contains configurations that are applicable to all units within that directory.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
+environment "an_account" {
+  filter {
+    paths = ["an-account/*"]
+  }
+
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+In this example, all the units located within the `an-account` directory sibling to the `.gruntwork` directory will assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands by Pipelines.
+
+A typical approach to building Pipelines configurations is to first define minimal configurations that address the most common use-cases, and then to refactor and generalize those configurations as needed to reduce repetition.
+
+More details on how these configurations are defined will be detailed below.
+
+## Global Configurations
+
+The configurations found within a `.gruntwork` directory found in the current working directory, or a parent directory of the current working directory are referred to as global configurations. These configurations are typically applicable within a wide range of contexts within a repository, and are the primary mechanism for configuring Pipelines.
+
+Pipelines will attempt to find exactly one directory named `.gruntwork` when it is attempting to discover configurations. It will not continue to search for configurations in parent directories once it finds a `.gruntwork` directory.
+
+Note that you will frequently see filenames for configurations within the `.gruntwork` directory that are named after the configuration block that they define. This is a common pattern, but not a requirement. Any `.hcl` file found within a `.gruntwork` directory will be parsed by Pipelines as one global configuration.
+
+### Environment Blocks
+
+Environment blocks are used to define configurations that are applicable to a specific environment within a repository.
+
+The label applied to an environment block is the name of the environment. This is a user-defined label for the environment, and must be globally unique.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
+environment "an_environment" {
+  filter {
+    paths = ["an-environment/*"]
+  }
```

Comment:
Similar comment on the paths example here, I would suggest paths that are not the same as the environment name.

Reply:
What would you prefer it be called? I think most folks _should_ name the directories for their environments to be very similar to the names of their environments.

Reply:
Thats a good point.